lib/ddtrace/writer.rb

```diff
@@ -72,20 +72,19 @@ def send_spans(traces, transport)
       # Inject hostname if configured to do so
       inject_hostname!(traces) if Datadog.configuration.report_hostname
 
-      # Send traces an get a response.
-      response = transport.send_traces(traces)
+      # Send traces and get responses
+      responses = transport.send_traces(traces)
 
-      unless response.internal_error?
-        @traces_flushed += traces.length unless response.server_error?
-
-        # Update priority sampler
-        unless priority_sampler.nil? || response.service_rates.nil?
-          priority_sampler.update(response.service_rates)
-        end
+      # Tally up successful flushes
+      responses.reject { |x| x.internal_error? || x.server_error? }.each do |response|
+        @traces_flushed += response.trace_count
       end
 
+      # Update priority sampler
+      update_priority_sampler(responses.last)
```

Comment:
What if there are no successful responses?

Reply:
I see `update_priority_sampling_sampler` might be handling this?

Reply:
You are correct, this just a matter of extracting a somewhat complex logic block into its own method.